src/App.tsx

```diff
@@ -1,73 +1,95 @@
-import React from 'react';
+import debounce from 'lodash.debounce';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { DropDown } from './components/DropDown';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [people] = useState<Person[]>(peopleFromServer);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  const inputValue = useRef<HTMLInputElement | null>(null);
+  const applyQuery = useCallback(debounce(setAppliedQuery, 500), []);
```

Comment:
The `debounce` function is used here without including `setAppliedQuery` in the dependency array. Although `setAppliedQuery` is stable in this context, it's a good practice to include it in the dependency array to prevent potential issues if the function changes in the future.